tmt/steps/discover/fmf.py

```diff
@@ -269,22 +324,77 @@ def assert_git_url(plan_name=None):
+                pass
 
         # Fetch and extract distgit sources
         if dist_git_source:
             try:
                 self.extract_distgit_source(
-                    self.testdir, self.testdir, self.get('dist-git-type'))
+                    git_root, sourcedir, self.get('dist-git-type'))
             except Exception as error:
                 raise tmt.utils.DiscoverError(
                     f"Failed to process 'dist-git-source'.", original=error)
 
+            # Check what should be extracted from the sources
+            if dist_git_extract:
+                if dist_git_extract == '/':
+                    dist_git_extract = sourcedir
+                else:
+                    try:
+                        dist_git_extract = glob.glob(os.path.join(
+                            sourcedir, dist_git_extract.lstrip('/')))[0]
+                    except IndexError:
+                        raise tmt.utils.DiscoverError(
+                            f"Couldn't glob '{dist_git_extract}' "
+                            f"within extracted sources.")
+
+            # Check sources for the fmf root, copy git root if not found
+            if not dist_git_init and not dist_git_extract:
+                try:
+                    top_fmf_root = tmt.utils.find_fmf_root(sourcedir)[0]
+                except tmt.utils.MetadataError:
+                    dist_git_extract = sourcedir
+                    if not dist_git_merge:
+                        self.warn(
+                            "Extracted sources do not contain fmf root, "
+                            "merging with plan data. Avoid this warning by "
+                            "explicit use of the '--dist-git-merge' option.")
+                        self.debug(f"Copy '{git_root}' to '{self.testdir}'.")
+                        if not self.opt('dry'):
+                            shutil.copytree(
+                                git_root, self.testdir, symlinks=True)
+
+            # Initialize or remove fmf root
+            if dist_git_init:
+                if not dist_git_extract:
+                    dist_git_extract = sourcedir
+                if not self.opt('dry'):
+                    fmf.Tree.init(dist_git_extract)
+            elif dist_git_remove_fmf_root:
+                if not self.opt('dry'):
+                    shutil.rmtree(os.path.join(
+                        dist_git_extract or top_fmf_root, '.fmf'))
+                if not dist_git_extract:
+                    dist_git_extract = sourcedir
+
+            # Now can safely default to top_fmf_root
+            if not dist_git_extract:
+                dist_git_extract = top_fmf_root
+
+            # Now copy dist_git_extract into tests
+            if not self.opt('dry'):
+                shutil.copytree(
+                    dist_git_extract,
+                    self.testdir,
+                    symlinks=True,
+                    dirs_exist_ok=True)
```

Comment:
Seems the `dir_exist_ok` has been added only in Python 3.8:
https://docs.python.org/3/library/shutil.html#shutil.copytree
So I guess this would not work on `el8`. Haven't tested though.